### ISSUE TITLE
[Bug]: `runtime error: invalid memory address or nil pointer dereference`

### ISSUE THREAD

**Issue by justinTM** [jetpack-io/devbox#1562]
```console
$ devbox add sass
Error: exit status 1

Error: runtime error: invalid memory address or nil pointer dereference
```

## device info
```console
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04 LTS
Release:        22.04
Codename:       jammy
```

```console
$ lshw
justins-bodhi
    description: Desktop Computer
    product: Peppy
    vendor: Acer
    version: 1.0
    serial: 123456789
    width: 64 bits
    capabilities: smbios-2.7 dmi-2.7 smp vsyscall32
    configuration: boot=normal chassis=desktop
  *-core
       description: Motherboard
       physical id: 0
          vendor: GenuineIntel
          physical id: 3
          version: Intel(R) Celeron(R) 2955U @ 1.40GHz
          configuration: cores=16
     *-memory
          description: System memory
          physical id: 1
          size: 4GiB
     *-cpu:1
          product: Intel(R) Celeron(R) 2955U @ 1.40GHz
          vendor: Intel Corp.
          physical id: 2
          bus info: cpu@0
          version: 6.69.1
          size: 1396MHz
          capacity: 1400MHz
          width: 64 bits
          capabilities: fpu fpu_exception wp vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp x86-64 constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg cx16 xtpr pdcm pcid sse4_1 sse4_2 movbe popcnt tsc_deadline_timer xsave rdrand lahf_lm abm cpuid_fault epb invpcid_single pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust erms invpcid xsaveopt dtherm arat pln pts md_clear flush_l1d cpufreq
          configuration: microcode=38
          description: Host bridge
          product: Haswell-ULT DRAM Controller
          vendor: Intel Corporation
          physical id: 100
          bus info: pci@0000:00:00.0
          version: 09
          width: 32 bits
          clock: 33MHz
          configuration: driver=hsw_uncore
          resources: irq:0
```

**Comment by savil** [comment]
@justinTM thank you for reporting. A few questions:

- What does `devbox version --verbose` return for you?
- Could you also please re-run the command with `DEVBOX_DEBUG=1 devbox add sass` and share the output?
- Does this happen for any package that you are trying to add (e.g. `devbox add hello`), or specifically with `sass` ?

**Comment by Omixxx** [comment]
"Hi, same here. 

` $ devbox shell
Error: exit status 1

Error: runtime error: invalid memory address or nil pointer dereference`

it happends also when i try to add a package. 
So i run `DEVBOX_DEBUG=1 devbox add sass` just to get a glance and the output was 
``

**Comment by savil** [comment]
@Omixxx thanks for sharing the error info! That was very helpful. 

I appears you ran into a Permission Denied error with `/nix/var/nix/daemon-socket/socket: Permission denied`, and then another error with our sentry telemetry code when Devbox CLI tried to log the first error.

1. The specific nil-pointer issue from `telemetry/sentry.go:55` was fixed in https://github.com/jetpack-io/devbox/pull/1569 and shipped in the Devbox 0.7.1 release on Friday.
2. The `Permission denied` error seems to indicate that for some reason your `nix` client is unable to communicate with the `nix` daemon. This indicates an issue with your nix installation. Did you recently install, or update your nix? To use Devbox, we don't expect users to ever have to invoke `sudo` _except_ when installing devbox, or installing nix.

**Comment by savil** [comment]
I'm going to close this specific issue about the `nil pointer dereference`, since I believe its fixed. However, please let us know if you encounter it again (feel free to open a new issue).
